tests/test_config.py

```diff
@@ -0,0 +1,111 @@
+from conftest import upstream_webdav_server, tmp_certificate
+from flask import url_for
+import pytest
+from typing import Any
+
+
+@pytest.mark.timeout(30)
+def test_valid_owncert_config(app: Any, client: Any):
+    with upstream_webdav_server():
+        with app.app_context():
+            ca_bundle, certificate = tmp_certificate(1)
+            open(app.config['CTADS_CABUNDLE'], 'w').write(ca_bundle)
+            r = client.post(url_for('upload_cert'), json={
+                'certificate': certificate})
+            assert r.status_code == 200
+
+
+@pytest.mark.timeout(30)
+def test_invalid_owncert_config(app: Any, client: Any):
+    with upstream_webdav_server():
+        with app.app_context():
+            _, certificate = tmp_certificate(1)
+            r = client.post(url_for('upload_cert'), json={
+                'certificate': certificate})
+            assert r.status_code == 400 and \
+                r.text == 'invalid certificate verification chain'
+
+
+@pytest.mark.timeout(30)
+def test_expired_owncert_config(app: Any, client: Any):
+    with upstream_webdav_server():
+        with app.app_context():
+            ca_bundle, certificate = tmp_certificate(-1)
+            r = client.post(url_for('upload_cert'), json={
+                            'certificate': certificate})
+            assert r.status_code == 400 and \
+                r.text == 'certificate expired'
```

Comment:
but it's also invalid chain here? invalid chain should take precedence over expired, since if we can not check certificate signature we can not trust anything in it, including validity time